sdk/src/main/java/com/akkaserverless/javasdk/action/Action.java

```diff
@@ -34,7 +40,37 @@ protected final ActionContext actionContext() {
   }
 
   /** INTERNAL API */
-  public final void setActionContext(Optional<ActionContext> context) {
+  public final void _internalSetActionContext(Optional<ActionContext> context) {
     actionContext = context;
   }
+
+  public final Effect.Builder effects() {
+    return ActionEffectImpl.Builder$.MODULE$;
+  }
+
+  /** @param <T> The type of the message that must be returned by this call. */
+  public interface Effect<T> {
+    interface Builder {
+      // FIXME should it rather be "Reply"?
+      <S> Effect<S> message(S message);
+
+      <S> Effect<S> forward(ServiceCall serviceCall);
+
+      <S> Effect<S> noReply();
+
+      <S> Effect<S> error(String description);
+
+      <S> Effect<S> asyncMessage(CompletionStage<S> message);
+
+      <S> Effect<S> asyncEffect(CompletionStage<Effect<S>> futureEffect);
+    }
+
+    boolean isEmpty();
+
+    Effect<T> addSideEffect(SideEffect sideEffect);
+
+    Effect<T> addSideEffects(Collection<SideEffect> sideEffects);
+
+    Collection<SideEffect> sideEffects();
```

Comment:
this can be removed, we don't have it in ValueEntity for example

Reply:
Same here, here for the TCK, I'll get rid of it.